txstratum/client.py

```diff
@@ -0,0 +1,111 @@
+"""
+Copyright (c) Hathor Labs and its affiliates.
+
+This source code is licensed under the MIT license found in the
+LICENSE file in the root directory of this source tree.
+"""
+
+from typing import List, NamedTuple, Optional
+from urllib.parse import urljoin
+
+from aiohttp import ClientSession
+from txstratum.commons import Block, TxOutput
+
+REQUIRED_HATHOR_API_VERSION = 'v1a'
+
+
+class BlockTemplate(NamedTuple):
+    """Block template."""
+
+    data: bytes
+    height: int
+
+    def to_dict(self):
+        """Return dict for json serialization."""
+        return {
+            'data': self.data.hex(),
+            'height': self.height,
+        }
+
+
+class HathorVersion(NamedTuple):
+    """Hathor backend version."""
+
+    major: int
+    minor: int
+    patch: int
+
+
+class HathorClient:
+    """Used to communicate with Hathor's full-node."""
+
+    USER_AGENT = 'tx-mining-service'
+
+    def __init__(self, server_url: str, api_version: str = REQUIRED_HATHOR_API_VERSION):
+        """Init HathorClient with a Hathor's full-node backend."""
+        self._base_url = urljoin(server_url, api_version).rstrip('/') + '/'
+        self._base_headers = {
+            'User-Agent': self.USER_AGENT,
+        }
+        self._session: Optional[ClientSession] = None
+
+    async def start(self):
+        """Start a session with the backend."""
+        self._session = ClientSession(headers=self._base_headers)
+
+    async def stop(self):
+        """Stop a session with the backend."""
+        await self._session.close()
+        self._session = None
+
+    def _get_url(self, url: str) -> str:
+        return urljoin(self._base_url, url.lstrip('/'))
+
+    async def version(self) -> HathorVersion:
+        """Return the version of the backend."""
+        assert self._session is not None
+        async with self._session.get(self._get_url('version')) as resp:
+            data = await resp.json()
+            ver = data['version']
+            major, minor, patch = ver.split('.')
+            return HathorVersion(int(major), int(minor), int(patch))
+
+    async def get_block_template(self) -> BlockTemplate:
+        """Return a block template."""
+        assert self._session is not None
+        async with self._session.get(self._get_url('get_block_template')) as resp:
+            data = await resp.json()
+            # TODO Get block's height.
+            height = 0
```

Comment:
Is this height that will come from the full node?

Reply:
It is. I know it is somewhere in the `data` object.

Reply:
It should be on `data.metadata.height`.

Reply:
Done. Thanks!